src/HH4b/processors/bbbbSkimmer.py

```diff
@@ -166,133 +172,134 @@ def process(self, events: ak.Array):
 
+        # JMSR variables
         for var in ["msoftdrop", "particleNet_mass"]:
             key = self.skim_vars["FatJet"][var]
             for shift, vals in jmsr_shifted_vars[var].items():
                 # overwrite saved mass vars with corrected ones
                 label = "" if shift == "" else "_" + shift
                 ak8FatJetVars[f"ak8FatJet{key}{label}"] = vals
 
-        # particlenet xbb vs qcd
-
-        fatjets["Txbb"] = fatjets.particleNetMD_Xbb / (
-            fatjets.particleNetMD_QCD + fatjets.particleNetMD_Xbb
-        )
-
-        ak8FatJetVars["ak8FatJetParticleNetMD_Txbb"] = pad_val(
-            fatjets["Txbb"],
-            num_jets,
-            axis=1,
-        )
-
         # dijet variables
-
-        dijetVars = {}
-
-        for shift in jec_shifted_vars["pt"]:
+        fatDijetVars = {}
+        for shift in jec_shifted_fatjetvars["pt"]:
             label = "" if shift == "" else "_" + shift
-            dijetVars = {**dijetVars, **self.getDijetVars(ak8FatJetVars, pt_shift=label)}
+            fatDijetVars = {**fatDijetVars, **self.getFatDijetVars(ak8FatJetVars, pt_shift=label)}
 
         for shift in jmsr_shifted_vars["msoftdrop"]:
             if shift != "":
                 label = "_" + shift
-                dijetVars = {
-                    **dijetVars,
-                    **self.getDijetVars(ak8FatJetVars, mass_shift=label),
+                fatDijetVars = {
+                    **fatDijetVars,
+                    **self.getFatDijetVars(ak8FatJetVars, mass_shift=label),
                 }
 
         otherVars = {
             key: events[var.split("_")[0]]["_".join(var.split("_")[1:])].to_numpy()
             for (var, key) in self.skim_vars["other"].items()
         }
 
-        skimmed_events = {**skimmed_events, **ak8FatJetVars, **dijetVars, **otherVars}
+        # flatten object variables
+        ak8FatJetVars = flatten_dict(ak8FatJetVars, "ak8FatJet")
+
+        skimmed_events = {**skimmed_events, **ak8FatJetVars, **fatDijetVars, **otherVars}
 
         ######################
         # Selection
         ######################
 
         # OR-ing HLT triggers
         if isData:
-            for trigger in HLTs[year]:
+            for trigger in self.HLTs[year]:
                 if trigger not in events.HLT.fields:
                     logger.warning(f"Missing HLT {trigger}!")
 
             HLT_triggered = np.any(
                 np.array(
-                    [events.HLT[trigger] for trigger in HLTs[year] if trigger in events.HLT.fields]
+                    [
+                        events.HLT[trigger]
+                        for trigger in self.HLTs[year]
+                        if trigger in events.HLT.fields
+                    ]
                 ),
                 axis=0,
             )
             add_selection("trigger", HLT_triggered, *selection_args)
 
-        # pt, eta cuts: check if jet passes pT cut in any of the JEC variations
```

Comment:
Are you removing the fatjet pT preselection?

Reply:
Oh right, yeah I didn't consider the different pre-selections for different analyses. Probably better then to modularize the object selection, pre-selection, and skimming, and add an option for which analysis is being run. (For future)